daemon-scheduler/pkg/deployment/environment_test.go

```diff
@@ -185,6 +186,139 @@ func (suite *EnvironmentTestSuite) TestGetEnvironment() {
 	assert.Exactly(suite.T(), suite.environment1, env, "Expected the environment to match the expected one")
 }
 
+func (suite *EnvironmentTestSuite) TestGetCurrentDeploymentInProgress() {
+	environment, err := types.NewEnvironment("TestGetCurrentDeployment", "td-arn", "cluster-arn")
+	assert.Nil(suite.T(), err, "Unexpected error when creating environment")
+	deployment, err := types.NewDeployment("td-arn", uuid.NewRandom().String())
+	assert.Nil(suite.T(), err, "Unexpected error when creating deployment")
+	deployment.Status = types.DeploymentInProgress
+	environment.InProgressDeploymentID = deployment.ID
+	environment.Deployments[deployment.ID] = *deployment
+
+	suite.environmentStore.EXPECT().GetEnvironment(suite.ctx, environment.Name).
+		Return(environment, nil)
+
+	d, err := suite.environment.GetCurrentDeployment(suite.ctx, environment.Name)
+	assert.Nil(suite.T(), err, "Unexpected error when calling GetCurrentDeployment")
+	assert.Exactly(suite.T(), deployment.ID, d.ID, "Expected the deployment to match the expected one")
+}
+
+func (suite *EnvironmentTestSuite) TestGetCurrentDeploymentPending() {
+	environment, err := types.NewEnvironment("TestGetCurrentDeployment", "td-arn", "cluster-arn")
+	assert.Nil(suite.T(), err, "Unexpected error when creating environment")
+	deployment, err := types.NewDeployment("td-arn", uuid.NewRandom().String())
+	assert.Nil(suite.T(), err, "Unexpected error when creating deployment")
+	deployment.Status = types.DeploymentPending
+	environment.PendingDeploymentID = deployment.ID
+	environment.Deployments[deployment.ID] = *deployment
+
+	suite.environmentStore.EXPECT().GetEnvironment(suite.ctx, environment.Name).
+		Return(environment, nil)
+
+	d, err := suite.environment.GetCurrentDeployment(suite.ctx, environment.Name)
+	assert.Nil(suite.T(), err, "Unexpected error when calling GetCurrentDeployment")
+	assert.Exactly(suite.T(), deployment.ID, d.ID, "Expected the deployment to match the expected one")
+}
+
+func (suite *EnvironmentTestSuite) TestGetCurrentDeploymentCompleted() {
+	environment, err := types.NewEnvironment("TestGetCurrentDeployment", "td-arn", "cluster-arn")
+	assert.Nil(suite.T(), err, "Unexpected error when creating environment")
+	deployment, err := types.NewDeployment("td-arn", uuid.NewRandom().String())
+	assert.Nil(suite.T(), err, "Unexpected error when creating deployment")
+	deployment.Status = types.DeploymentCompleted
+	environment.InProgressDeploymentID = deployment.ID
```

Comment:
it is done by the monitor, but there can be a race. this makes it clear that even this case is handled